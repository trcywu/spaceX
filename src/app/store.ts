import { configureStore } from '@reduxjs/toolkit'
import rootReducer from './rootReducer'

const createStore = () => {
    const store = configureStore({
        reducer: rootReducer,
    })
    if (process.env.NODE_ENV === 'development' && module.hot) {
        module.hot.accept('./rootReducer', () => {
            const newRootReducer = require('./rootReducer').default
            store.replaceReducer(newRootReducer)
        })
    }
    return store
}

export default createStore
