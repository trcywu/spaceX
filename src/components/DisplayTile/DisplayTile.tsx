import React, { FC } from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'

type Props = {
    name: string
    type: string
    flickrImg: string
    active: boolean
    id: string
    pageName: string
}

const StyledTileContainer = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-self: stretch;
    font-size: 12px;
    border: 2px solid black;
    border-radius: 2px;
    padding: 20px;
    margin-right: 10px;
`

const StyledImgContainer = styled.img`
    width: 100%;
    height: 400px;
`

const StyledDescriptor = styled.span`
    font-size: 14px;
    padding: 5px 0 5px 0;
    justify-content: space-between;
    text-transform: capitalize;
`

const StyledMoreInfoWrapper = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: space-between;
`
const DisplayTile: FC<Props> = ({
    name,
    type,
    flickrImg,
    active,
    id,
    pageName,
}) => {
    return (
        <StyledTileContainer data-testid="display-tile" aria-labelledby={name}>
            <StyledImgContainer src={flickrImg} alt={name} width="100%" />
            <StyledDescriptor>Name: {name}</StyledDescriptor>
            <StyledDescriptor>Type: {type}</StyledDescriptor>
            <StyledMoreInfoWrapper>
                <StyledDescriptor>
                    Active: {active ? 'yes' : 'no'}
                </StyledDescriptor>
                <StyledDescriptor>
                    <Link to={`/${pageName}/${id}`}>More Info</Link>
                </StyledDescriptor>
            </StyledMoreInfoWrapper>
        </StyledTileContainer>
    )
}

export default DisplayTile
