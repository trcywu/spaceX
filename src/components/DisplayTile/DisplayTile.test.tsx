import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import DisplayTile from './DisplayTile'
import store from 'app/store'
import { MemoryRouter } from 'react-router-dom'

const testProps = {
    name: 'testName',
    type: 'testType',
    flickrImg: 'www.test.com',
    active: true,
    id: '456',
    pageName: 'testPage',
}

test('render tile', () => {
    const mockStore = store()
    const { asFragment } = render(
        <Provider store={mockStore}>
            <MemoryRouter>
                <DisplayTile {...testProps} />
            </MemoryRouter>
        </Provider>
    )
    expect(asFragment()).toMatchSnapshot()
})
