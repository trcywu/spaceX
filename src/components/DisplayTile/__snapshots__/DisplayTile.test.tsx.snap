// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render tile 1`] = `
<DocumentFragment>
  <div
    aria-labelledby="testName"
    class="sc-bdfBwQ gtCuCm"
    data-testid="display-tile"
  >
    <img
      alt="testName"
      class="sc-gsTCUz fHwxFn"
      src="www.test.com"
      width="100%"
    />
    <span
      class="sc-dlfnbm ljbMun"
    >
      Name: testName
    </span>
    <span
      class="sc-dlfnbm ljbMun"
    >
      Type: testType
    </span>
    <div
      class="sc-hKgILt hJaAuR"
    >
      <span
        class="sc-dlfnbm ljbMun"
      >
        Active: yes
      </span>
      <span
        class="sc-dlfnbm ljbMun"
      >
        <a
          href="/testPage/456"
        >
          More Info
        </a>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
